src/detect-content.c

```diff
@@ -379,20 +379,33 @@ _Bool DetectContentPMATCHValidateCallback(const Signature *s)
     if (!(s->flags & SIG_FLAG_DSIZE)) {
         return TRUE;
     }
-
     int max_right_edge_i = SigParseGetMaxDsize(s);
     if (max_right_edge_i < 0) {
         return TRUE;
     }
 
     uint32_t max_right_edge = (uint32_t)max_right_edge_i;
+    int content_temp_length = 0;
 
     const SigMatch *sm = s->init_data->smlists[DETECT_SM_LIST_PMATCH];
     for ( ; sm != NULL; sm = sm->next) {
         if (sm->type != DETECT_CONTENT)
             continue;
         const DetectContentData *cd = (const DetectContentData *)sm->ctx;
         uint32_t right_edge = cd->content_len + cd->offset;
+
+        int distance_size = cd->distance;
+        int previous_content_length = content_temp_length;
+
+        if (cd->flags == DETECT_CONTENT_NEGATED) {
```

Comment:
This is likely too specific. What if NOCASE is also set? This should be ```if (cd->flags & DETECT_CONTENT_NEGATED) {``` I think.

Reply:
Thanks! I updated to use &

I will add some additional test coverage to see about the nocase or other keyword impact.